File: android_client/app/src/main/java/com/zigzag/client_app/model/Entity.java
package com.zigzag.client_app.model;

/**
 * The base class for all model objects.
 * <p>
 * Contains an unique identifier for the object, up to its own name space.
 * <p>
 * Entities are immutable.
 */
public class Entity {
    /**
     * The identifier for the entity.
     */
    private final EntityId id;

    /**
     * Construct an entity with a given id.
     *
     * @param id The id to use for the entity.
     */
    protected Entity(EntityId id) {
        this.id = id;
    }

    /**
     * Retrieve the id of the entity.
     *
     * @return The id of the entity.
     */
    public EntityId getId() {
        return id;
    }
}

File: ios_client/ios_client/models/Entity.swift
/**
 * Entity.swift
 * Created by Liviu Coman on 3/24/15.
 * Copyright (c) 2015 LiviuComan. All rights reserved.
 */

import UIKit

/**
 * The base class for all model objects.
 * <p>
 * Contains an unique identifier for the object, up to its own name space.
 * <p>
 * Entities are immutable.
 */
class Entity: NSObject {
    /**
     * Declare the Entity's identifier (which is an explicitly initialized
     * optional, since an implicit optional would be too much hassle)
     */
    private var id: EntityId! = nil
    
    /**
     * Swift constructor, named init()
     *
     * @param: the Entity id object
     */
    init(id: EntityId) {
        self.id = id;
    }
    
    /**
     * Retrieve the id of the entity.
     *
     * @return The id of the entity.
     */
    func getId() -> EntityId {
        return self.id;
    }
}
